apps/client/src/app/other-resources/other-resources-result/other-resources-result.component.ts

```diff
@@ -0,0 +1,212 @@
+            }
+
+            setTimeout(() => { this.message = ''; }, 3000);
+          }
+           catch (ex) {
+            this.message = 'The content you have specified is invalid: ' + ex.message;
+            //Deactivate save button when true
+            this.serializationError = true;
+          }
+        }
+      });
+  }
+
+  contentHasChanged(newValue: string) {
+    this.content = newValue;
+    this.contentChanged.emit();
+  }
+
+  ngOnInit() {
+    this.message = 'Opening resource';
+
+    this.fhirService.read(this.route.snapshot.params.type, this.route.snapshot.params.id)
+      .subscribe((results: DomainResource) => {
+
+        this.data = new OpenedResource(results);
+        this.content = JSON.stringify(this.data.resource, null, '\t');
+        this.validation = this.fhirService.validate(this.data.resource);
+
+        setTimeout(() => {
+          this.message = 'Resource opened.';
+        }, 100);
+      }, (err) => {
+        this.message = 'Error opening resource: ' + getErrorString(err);
+      });
+  }
+
+  changeType() {
+    console.log('Changing content type');
+    switch (this.selected) {
+      case 'JSON':
+        this.content = JSON.stringify(this.data.resource, null, '\t');
+        break;
+      case 'XML':
+        this.content = this.fhirService.serialize(this.data.resource);
+        break;
+    }
+  }
+
+  changeSubTab(event: NgbTabChangeEvent) {
+    this.data.activeSub = <any> event.nextId;
+  }
+
+  public downloadFile() {
+    const openedResource = this.data;
+    let type: string = this.selected;
+    switch (type) {
+      case 'XML':
+        const xml = this.fhirService.serialize(openedResource.resource);
+        const xmlBlob = new Blob([xml], {type: 'application/xml'});
+        saveAs(xmlBlob, openedResource.resource.id + '.xml');
+        break;
+      case 'JSON':
+        const json = JSON.stringify(openedResource.resource, null, '\t');
+        const jsonBlob = new Blob([json], {type: 'application/json'});
+        saveAs(jsonBlob, openedResource.resource.id + '.json');
+        break;
+    }
+  }
+
+  public uploadFile(event: any) {
+    let type: string = this.selected;
+    const reader = new FileReader();
+    const openedResource = this.data;
+
+    if (event.target.files && event.target.files.length > 0) {
+      const file = event.target.files[0];
+      reader.onload = () => {
+        const content = <string>reader.result;
+        let resource: DomainResource;
+
+        switch (type) {
+          case 'json':
+            resource = JSON.parse(content);
+            break;
+          case 'xml':
+            resource = this.fhirService.deserialize(content);
+            break;
+          default:
+            throw new Error('Unexpected type specified: ' + type);
+        }
+
+        this.message = 'Updating the resource';
+
+        this.fhirService.update(resource.resourceType, openedResource.resource.id, resource)
```

Comment:
trying to think of what happens when the user has made changes in the text area, but hasn't saved the changes yet, and then uploads a file... perhaps the behavior would be better expected/understood if when the user uploads a file, the file content is reflected in the text area, but not immediately persisted/saved? What do you think?

Reply:
Hm... I'm inclined to agree actually.  If we keep consistent with how everything else works here, we should only have the text content update rather than the server stored version of the text so that we only interact with the server when we press save. I'll see what I can do to modify this appropriately.